backend/migrations/versions/075d344ae2cc_mf_triggers.py

```diff
@@ -87,6 +87,25 @@ def upgrade():
         FOR EACH ROW
         EXECUTE FUNCTION dataset_delete_trigger_function();
     """)
+
+    op.execute("""
+        CREATE OR REPLACE FUNCTION metadata_form_delete_trigger_function()
+        RETURNS TRIGGER AS $$
+        BEGIN
+            -- Delete from resource_permission_policy
+            DELETE FROM resource_policy_permission
+            WHERE "sid" in (SELECT sid from resource_policy where "resourceUri"=OLD.uri and "resourceType"='MetadataForm');
+            DELETE FROM resource_policy where "resourceUri"=OLD.uri;
+            RETURN OLD;
+        END;
+        $$ LANGUAGE plpgsql;
+
+        -- Create the trigger for dataset table
+        CREATE TRIGGER metadata_form_delete_trigger
+        BEFORE DELETE ON metadata_form
+        FOR EACH ROW
+        EXECUTE FUNCTION metadata_form_delete_trigger_function();
+    """)
```

Comment:
is this trigger used anywhere? Do we grant permissions on Metadata Form resources? 

From what I see we only grant permissions to CREATE or ATTACH Form on the parent resource (i.e. env, org, etc.) and then for UPDATE, DELETE, or similar on form itself we just check if owner or not